eos-tech-support/eos-download-image

```diff
@@ -1,125 +1,114 @@
-EOF
-}
-
-while true; do
-    case "$1" in
-        -v|--version)
-            shift
-            VERSION=$1
-            shift
-            ;;
-        -p|--personality)
-            shift
-            PERSONALITY=$1
-            shift
-            ;;
-        -r|--product)
-            shift
-            PRODUCT=$1
-            shift
-            ;;
-        -i|--internal)
-            shift
-            BASE_URL="$INTERNAL_BASE_URL"
-            ;;
-        -h|--help)
-            usage
-            exit 0
-            ;;
-        --)
-            shift
-            break
-            ;;
-    esac
-done
-
-log() {
-    echo >&2
-    echo "$@" >&2
-}
-
-download_and_verify() {
-  log "Downloading $1 from $2"
-  wget --continue $2 || return 1
-
-  log "Downloading signature for $1 from $3"
-  wget --continue $3 || return 1
-
-  ensure_keyring
-
-  log "Verifying..."
-  gpgv --keyring $KEYRING $(basename $3) $(basename $2) || return 1
-
-  log "Download ok."
-  return 0
-}
-
-ensure_keyring() {
-  if [ -f "$SYSTEM_KEYRING" ]
-  then
-    KEYRING="$SYSTEM_KEYRING"
-  else
-    KEYRING="$(readlink -f ./$KEYRING_FILENAME)"
-    if [ ! -f "$KEYRING" ]
-    then
-      log "Fetching Endless keyring"
-      wget -O "$KEYRING" "$KEYRING_URL"
-    fi
-  fi
-}
-
-log "Fetching manifest..."
-JSON=$(wget -O- $MANIFEST_BASE_URL/releases-$PRODUCT.json.gz | gzip -d)
-
-# If the user specified a version, select that but otherwise we go with latest
-if test -z "$VERSION"
-then
-  SELECTED=$(echo $JSON | jq '.images[] | values' | jq -s 'max_by(.version)')
-else
-  SELECTED=$(echo $JSON | jq '.images[] | values' | jq -s ".[] | select(.version | contains(\"$1\"))")
-fi
-
-if test -z "$PERSONALITY"
-then
-  PERSONALITY="base"
-fi
-
-# Only full images are supported for now
-OS_IMAGE_URL=$(echo $SELECTED | jq -r ".personality_images[\"$PERSONALITY\"] | .full | .file")
-OS_SIGNATURE_URL=$(echo $SELECTED | jq -r ".personality_images[\"$PERSONALITY\"] | .full | .signature")
-
-if test "null" = "$OS_IMAGE_URL" -o "null" = $OS_SIGNATURE_URL
-then
-  log "Problem fetching the image manifest. Please try again."
-  exit 1
-fi
-
-# Download & verify
-download_and_verify "image" $BASE_URL/$OS_IMAGE_URL $BASE_URL/$OS_SIGNATURE_URL >&2
-
-echo $(basename $OS_IMAGE_URL)
+#!/usr/bin/env python3
+# vim: fileencoding=utf-8 sts=4 sw=4 et
+import argparse
+import codecs
+import distutils.version
+import gzip
+import json
+import os
+import pipes
+import subprocess
+import sys
+import urllib.request
+
+INTERNAL_BASE_URL = "http://images.endlessm-sf.com"
+PUBLIC_BASE_URL = "https://d1anzknqnc1kmb.cloudfront.net"
+KEYRING_FILENAME = "eos-image-keyring.gpg"
+KEYRING_URL = PUBLIC_BASE_URL + "/" + KEYRING_FILENAME
+KEYRING_SYSTEM_PATH = "/usr/share/keyrings/eos-image-keyring.gpg"
+
+
+def log(*args):
+    sys.stderr.write('\n')
+    print(*args, file=sys.stderr)
+    sys.stderr.flush()
+
+
+def check_call(*args):
+    log('$ ' + ' '.join(map(pipes.quote, args)))
```

Comment:
Ah, `pipes.quote` is [apparently](https://docs.python.org/2.7/library/pipes.html#pipes.quote) a Python 2-ism. Thanks!